sdk/dapp-kit/src/components/WalletProvider.tsx

```diff
@@ -24,7 +24,7 @@ interface WalletProviderProps {
 	storageKey?: string;
 
 	/** A list of features that are required for the dApp to function. This filters the list of wallets presented to users when selecting a wallet to connect from, ensuring that only wallets that meet the dApps requirements can connect. */
-	requiredFeatures?: string[];
+	requiredFeatures?: (keyof WalletWithRequiredFeatures['features'])[];
```

Comment:
Doesn't this mean we lose the ability to define custom features?

Reply:
I specified `IdentifierRecord<unknown>` on the `WalletWithRequiredFeatures` type to account for custom features, so TypeScript ends up evaluating the `requiredFeatures` prop as `${string}:${string}`:

<img width="751" alt="image" src="https://github.com/MystenLabs/sui/assets/7453188/b26c5983-8d10-45d4-ae22-2acc9e5cc02a">


```
type WalletWithRequiredFeatures = WalletWithFeatures<
	MinimallyRequiredFeatures &
		Partial<SuiFeatures> &
		Partial<StandardDisconnectFeature> &
		**IdentifierRecord<unknown>**
>;
```